compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -127,7 +127,9 @@ static void addBufferizePasses(OpPassManager &passManager) {
 
 static void addTileAndDistributePasses(
     OpPassManager &pm, bool useFuseTensorPadWithConsumerPass = true) {
-  pm.addPass(createTileAndDistributeToWorkgroupsPass());
+  pm.addPass(createTileAndDistributeToWorkgroupsPass(
+      /*maxWorkgroupParallelDims=*/1,
```

Comment:
I dont think we want to set this to 1 yet..... Can we leave it to the default 3....

Reply:
Leaving it to three has the same issue. The distribution loops are not removed and it regresses the total dispatch sizes. That's why I set it to one, which removes the loops when doing distribution.

![image](https://user-images.githubusercontent.com/6949191/235265602-abd105b8-7cc9-491c-84bd-3918f9d42d31.png)

Reply:
Should I remove those `RemoveSingleIterationLoop` passes with the PR as well?

Reply:
You should be able to set it to 3, and also use `linalg::DistributionMethod::CylciNumProcsEqNumIters`.

You can remove that pass from the CPU backends. I think VMVX still needs that pass (mostly because it is also relying on some loop canonicalizations that get brought in from that pass I think, we can look into that later)

Reply:
Setting it to 3 triggers other issues. We won't get static shapes, and they don't get vectorized. This regresses both execution time and total dispatch sizes.

IR dump before and after vectorization w/ setting to 3: https://gist.githubusercontent.com/hanhanW/3f1cb8797ae44c28b06189e657f210a4/raw

IR dump before and after vectorization  w/ setting to 1: https://gist.githubusercontent.com/hanhanW/eb93102ed10077bc5ea949be38828c9c/raw

Reply:
I filed an issue for this: https://github.com/openxla/iree/issues/13466